packages/presentation/src/file.ts

```diff
@@ -61,6 +119,42 @@ export function getUploadUrl (): string {
   return template.replaceAll(':workspace', encodeURIComponent(getCurrentWorkspaceId()))
 }
 
+function getUploadConfig (): UploadConfig {
+  return getMetadata<UploadConfig>(plugin.metadata.UploadConfig) ?? { 'form-data': { url: getUploadUrl() } }
+}
+
+function getFileUploadMethod (blob: Blob): { method: FileUploadMethod, url: string } {
+  const config = getUploadConfig()
+
+  const signedUrl = config['signed-url']
+  if (signedUrl !== undefined && signedUrl.size < blob.size) {
+    return { method: 'signed-url', url: signedUrl.url }
+  }
+
+  return { method: 'form-data', url: config['form-data'].url }
+}
+
+/**
+ * @public
+ */
+export function getFileUploadParams (blobId: string, blob: Blob): FileUploadParams {
+  const workspaceId = encodeURI(getCurrentWorkspaceId())
+  const fileId = encodeURI(blobId)
```

Comment:
Minor: is the use of `encodeURI` instead of `encodeURIComponent` intended?

Reply:
nope, that's a bug

Reply:
resolved